e3sm_diags/driver/utils/regrid.py

```diff
@@ -424,25 +422,34 @@ def _add_mask_for_2d(
         The variable key.
     tool : str
         The regridding tool being used.
-    dims_to_check : List[str]
-        A list of dimensions to check for skipping mask creation.
 
     Returns
     -------
     xr.Dataset
         The modified dataset with the mask variable added if applicable.
     """
-    var_dims = ds[var_key].dims
+    ds_new = ds.copy()
+    var = ds_new[var_key]
+    var_dims = var.dims
 
-    if tool == "regrid2" or not any(dim in dims_to_check for dim in var_dims):
+    spatial_dims = {"x", "y", "lon", "lat", "longitude", "latitude"}
+    is_spatial_var = len(var_dims) == 2 and all(
+        str(dim).lower() in spatial_dims for dim in var_dims
+    )
+
+    if tool == "regrid2" or is_spatial_var:
         logger.debug(f"Creating mask for {var_key} with dimensions {var_dims}")
-        ds["mask"] = xr.where(~np.isnan(ds[var_key]), 1, 0)
+
+        if "mask" in ds_new:
+            logger.warning("Overwriting existing 'mask' variable in the dataset.")
+
+        ds_new["mask"] = xr.where(~np.isnan(var), 1, 0)
     else:
         logger.debug(
             f"Skipping mask creation for variable {var_key} with dimensions {var_dims}"
         )
 
-    return ds
+    return ds_new
```

Comment:
I updated the logic of `_add_mask` to make it easier to understand. Now, it checks if the variable is a 2D spatial variable (if it is, it allows regridding using xESMF).

This is simpler than the old approach, which checked whether *any* of the variable’s dimensions weren't in a list of vertical and time dimensions. That method was harder to follow and skipped regridding with xESMF if any non-spatial dimension was present.

Reply:
Also, it raises a `logger.warning` is an existing `mask` variable will be overwritten